tartufo/scanner.py

```diff
@@ -520,9 +550,11 @@ def scan(self) -> Generator[Issue, None, None]:
         # expectation that the write to _completed at the bottom of the critical
         # section cannot be reordered to appear after the implicit release of
         # _scan_lock (when viewed from a competing thread).
+        self.logger.debug("Waiting for scan lock")
         with self._scan_lock:
             if self._completed:
-                yield from self._issues
+                self.logger.debug("Scan already completed")
+                yield from self.load_issues()
```

Comment:
What about cases where the issues are not buffered to file?

Reply:
Also, if we have a max buffer size of 10,000 but then we encounter 10,050 issues... the first 10,000 get pushed into the file, but what happens to those remaining 50? I can't see how those ones are ever loaded and reported.

Reply:
If you see the end of load_issues, after it finishes the file, it yields from _issue_list:

`            except EOFError:
                self.logger.debug("pickle.load raised EOFError, exiting")
                yield from self._issue_list
                return`